### ISSUE TITLE
Videos that invoke AdvancedYoutubeVideoURLExtractor stopped working

### ISSUE THREAD

**Issue by azihassan** [azihassan/youtube-d#89]
The regex that matches the signatureCipher decoding stopped working:

```bash
$ youtube-d https://youtu.be/dQw4w9WgXcQ --verbose
Verbose mode : true
Handling https://youtu.be/dQw4w9WgXcQ
Cache hit (/tmp/dQw4w9WgXcQ.html), skipping HTML download...
base.js cache hit (/tmp/5b77d519.js), skipping download...
Downloaded video HTML
Attempt to dethrottle : Yes
Rick Astley - Never Gonna Give You Up (Official Music Video)
Rick Astley - Never Gonna Give You Up (Official Music Video)-dQw4w9WgXcQ-18.mp4
/tmp/Rick Astley - Never Gonna Give You Up (Official Music Video)-dQw4w9WgXcQ-18.mp4
Failed to match regular expression a\s*=\s*a\.split\(""\);\s*((.|\s)*?);\s*return a\.join\(""\)
source/helpers.d:88 immutable(char)[] helpers.matchOrFail(std.regex.Captures!(immutable(char)[]).Captures, immutable(char)[]) [0x562de0ea16ea]
source/helpers.d:74 immutable(char)[] helpers.matchOrFail!("a\s*=\s*a\.split\(""\);\s*((.|\s)*?);\s*return a\.join\(""\)", false).matchOrFail(immutable(char)[]) [0x562de0ea522d]
source/parsers.d:500 ref parsers.EncryptionAlgorithm parsers.EncryptionAlgorithm.__ctor(immutable(char)[], helpers.StdoutLogger) [0x562de0ea8f77]
source/parsers.d:306 immutable(char)[] parsers.AdvancedYoutubeVideoURLExtractor.getURL(int, bool) [0x562de0ea8689]
source/app.d:145 void app.handleURL(immutable(char)[], int, helpers.StdoutLogger, bool, bool, bool, bool, bool, bool, bool) [0x562de0e0b81d]
source/app.d:100 _Dmain [0x562de0e0b28a]
```

**Comment by azihassan** [comment]
New function:

```javascript
gMP=function(m){m=m.split("");Aw.t2(m,35);Aw.m8(m,3);Aw.Zy(m,50);Aw.t2(m,34);Aw.t2(m,40);Aw.Zy(m,24);Aw.m8(m,3);Aw.Zy(m,31);return m.join("")};
```

Like with [yesterday's bug](https://github.com/azihassan/youtube-d/issues/87#issuecomment-2529706933), the variable names are now randomized. The same change is applied to the functions themselves:

```javascript
var Aw={m8:function(m,e){m.splice(0,e)},
t2:function(m,e){var Z=m[0];m[0]=m[e%m.length];m[e%m.length]=Z},
Zy:function(m){m.reverse()}}
```

**Comment by azihassan** [comment]
The regex change I applied, which consisted of replacing parts like `function\(a,b\)` with `function\(\w,\w\)`, turned out to be too ambitious because it ate all my RAM at compile time. It took me a while to link this change to the compiler RAM usage, I completely forgot that I was using [ctRegex](https://dlang.org/phobos/std_regex.html#ctRegex) in the code. I ended up using dots (function\(.,.\)) to match variable names since I noticed they're single letters, but I'm not sure if it's going to be as robust.
